tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenTestFactory.cs

```diff
@@ -15,53 +15,78 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
 
 namespace MongoDB.Driver.Tests.JsonDrivenTests
 {
     public class JsonDrivenTestFactory
     {
         // private fields
         private readonly IMongoClient _client;
-        private readonly IMongoDatabase _database;
-        private readonly IMongoCollection<BsonDocument> _collection;
-        private readonly Dictionary<string, IClientSessionHandle> _sessionMap;
+        private readonly string _databaseName;
+        private readonly string _collectionName;
+        private readonly Dictionary<string, object> _objectMap;
 
         // public constructors
-        public JsonDrivenTestFactory(IMongoClient client, IMongoDatabase database, IMongoCollection<BsonDocument> collection, Dictionary<string, IClientSessionHandle> sessionMap)
+        public JsonDrivenTestFactory(IMongoClient client, string databaseName, string collectionName, Dictionary<string, object> objectMap)
         {
             _client = client;
-            _database = database;
-            _collection = collection;
-            _sessionMap = sessionMap;
+            _databaseName = databaseName;
+            _collectionName = collectionName;
+            _objectMap = objectMap;
         }
 
         // public methods
-        public JsonDrivenClientTest CreateTest(string name)
+        public JsonDrivenTest CreateTest(string receiver, string name)
         {
-            switch (name)
+            if (receiver.StartsWith("session"))
```

Comment:
"StartsWith" instead of "==" because the receiver might be "session0" or "session1".